src/Components/GhostChat.jsx

```diff
@@ -19,12 +19,12 @@ const GhostChat = ({
     });
   }, [socket]);
 
-  const playerRole =
-  playerInfo
-    ?.find(player => player?.player_id === playerState?.player_id || null)
+  const player = playerInfo?.find(
+    (player) => player?.player_id === playerState?.player_id || null
+  );
 
   const handleMessageSubmit = (event) => {
-    if (newMessage.length > 0) {
+    if (newMessage.length > 0 && (player.role === 1 || player.role === 3)) {
```

Comment:
I would suggest && player.role % 2 === 1 to handle the new roles currently being added

Reply:
on it